actor/src/main/scala/org/apache/pekko/util/UniqueRandomShortProvider.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, derived from Akka.
+ */
+
+package org.apache.pekko.util
+
+import org.apache.pekko.annotation.InternalApi
+
+import java.security.SecureRandom
+
+/**
+ * An efficient collision free threadsafe generator of pseudo random
+ * shorts based on https://en.wikipedia.org/wiki/Double_hashing#Enhanced_double_hashing
+ * using SecureRandom as a seed.
+ *
+ * The avoidance of collisions is achieved by using a Short Array buffer
+ * that is prefilled with a dimension of 65536 (~ 131072 bytes in size per
+ * instance of [[UniqueRandomShortProvider]]). This means it will loop
+ * through all possible 65536 values before starting to re-generating
+ * the same numbers again.
+ */
+
+@InternalApi
+private[pekko] class UniqueRandomShortProvider {
+  import UniqueRandomShortProvider._
+
+  @volatile private var index = 0L
+  @volatile private var increment = 0L
+  @volatile private var count = 0L
+  @volatile private var limit = 0
+  private val buffer: Array[Short] = new Array[Short](UniqueRandomShortProvider.MaxLimit)
+  val r = new SecureRandom()
+
+  resetBuffer()
+  resetSeed()
+
+  count = 1
+
+  private def resetSeed(): Unit = {
+    index = r.nextLong
+    increment = r.nextLong
+  }
+
+  private def resetBuffer(): Unit = {
+    System.arraycopy(Filler, 0, buffer, 0, UniqueRandomShortProvider.MaxLimit)
+    limit = UniqueRandomShortProvider.MaxLimit
+  }
+  def nextId(): Short = {
```

Comment:
Formatting is important, but PR comments on formatting (in large projects) don't scale.

Scalafmt is enabled on the project. I think it's Scalafmt's job to reformat and complain. Its configuration is never perfect, but it avoids the bike shedding in PRs. Does Scalafmt not add/require these new lines? If not, then maybe a configuration change would be needed.

I see there's also a `verifyCodeStyle` command in `build.sbt`, it would be a good idea to include this in CI to have the build fail if the code style fails.

Reply:
There is CI checking - forcing a newline before a new function is just not one of the rules that our scalafmt enforces. 

I agree about keeping scalafmt conversations out of PRs when there is automated checking of the agreed formatting already. Separate conversations can be had about changing the format rules.